mangoPURE/models/modules.py

```diff
@@ -0,0 +1,53 @@
+from .base import Embedder, TimedHead, TimedLoss, TimedModel
+from transformers import WhisperModel
+import torch
+from typing import Dict
+
+
+class WhisperEmbedder(Embedder):
+    def __init__(self, hf_name: str):
+        super().__init__()
+        self.model = WhisperModel.from_pretrained(hf_name).encoder
+
+    def forward(self, batch: Dict) -> Dict:
+        """
+        :param batch: dictionary that should have key "input_features"
+            that whisper encoder takes
+        :return: batch with whisper embedder output ("embeddings" ket in batch)
+        """
+        batch["embeddings"] = self.model(input_features=batch["input_features"]).last_hidden_state
+        del batch["input_features"]
```

Comment:
why? It can break other modules that use the same batch. Model should not break it